### PR TITLE
Wb/seq intergrating main

### PR DESCRIPTION
**Pull request by bradyjoestar**

### CODE REVIEW

**Review by wukongcheng — APPROVED** [review]

File: l2geth/rollup/sync_service.go [hunk 1]
```diff
@@ -919,12 +919,14 @@ func (s *SyncService) applyTransactionToTip(tx *types.Transaction, sequencer com
 
 	select {
 	case err := <-errCh:
+		fmt.Println("========we have receiverd errCh-------------------")
```
Comment (HaoyangLiu) [inline l2geth/rollup/sync_service.go:922]: `fmt.Println` -> `log.Debug`

File: l2geth/rollup/sync_service.go [hunk 2]
```diff
@@ -919,12 +919,14 @@ func (s *SyncService) applyTransactionToTip(tx *types.Transaction, sequencer com
...
 		log.Error("Got error waiting for transaction to be added to chain", "msg", err)
 		s.SetLatestL1Timestamp(ts)
 		s.SetLatestL1BlockNumber(bn)
 		s.SetLatestIndex(index)
 		return err
 	case <-s.chainHeadCh:
+		fmt.Println("========we have receiverd chainHeadCh-------------------")
```
Comment (HaoyangLiu) [inline l2geth/rollup/sync_service.go:929]: `fmt.Println` -> `log.Debug`

File: l2geth/internal/ethapi/api.go [hunk 3]
```diff
@@ -659,10 +659,10 @@ func (s *PublicBlockChainAPI) GetHeaderByHash(ctx context.Context, hash common.H
...
 
 // GetBlockByNumber returns the requested canonical block.
-// * When blockNr is -1 the chain head is returned.
-// * When blockNr is -2 the pending chain head is returned.
-// * When fullTx is true all transactions in the block are returned, otherwise
-//   only the transaction hash is returned.
+//   - When blockNr is -1 the chain head is returned.
+//   - When blockNr is -2 the pending chain head is returned.
+//   - When fullTx is true all transactions in the block are returned, otherwise
+//     only the transaction hash is returned.
```
Comment (HaoyangLiu) [inline l2geth/internal/ethapi/api.go:665]: Can we just revert the changes to this file?

File: l2geth/eth/backend.go [hunk 4]
```diff
@@ -225,6 +225,7 @@ func New(ctx *node.ServiceContext, config *Config) (*Ethereum, error) {
 	eth.miner.SetExtra(makeExtraData(config.Miner.ExtraData))
 
 	log.Info("Backend Config", "max-calldata-size", config.Rollup.MaxCallDataSize, "gas-limit", config.Rollup.GasLimit, "is-verifier", config.Rollup.IsVerifier, "using-bvm", rcfg.UsingBVM)
+
```
Comment (HaoyangLiu) [inline l2geth/eth/backend.go:228]: We can just revert the change to this file

File: l2geth/consensus/clique/synchronizer/server.go [hunk 5]
```diff
@@ -64,7 +67,14 @@ func (sync *Synchronizer) GetSchedulerAddr() (common.Address, error) {
 	if err != nil {
 		return common.Address{}, fmt.Errorf("create rpcClient failed: %s", err)
 	}
-	return seqContractInst.Scheduler(nil)
+	schedulerAddr, err := seqContractInst.Scheduler(nil)
+	for schedulerAddr.String() == "0x0000000000000000000000000000000000000000" && err == nil {
```
Comment (HaoyangLiu) [inline l2geth/consensus/clique/synchronizer/server.go:71]: We can use `common.Address{}.toString()` to replace `"0x0000000000000000000000000000000000000000"`

File: l2geth/consensus/clique/synchronizer/server.go [hunk 6]
```diff
@@ -88,9 +98,20 @@ func (sync *Synchronizer) GetSequencerSet() (SequencerSequencerInfos, error) {
 	}
 	// get sequencers from sequencer contract
 	var seqInfos SequencerSequencerInfos
-	seqInfos, err = seqContractInst.GetSequencers(nil)
-	if err != nil {
-		return nil, err
+	for i := 0; i < GET_SEQUENCER_RETRY_TIMES; i++ {
+		seqInfos, err = seqContractInst.GetSequencers(nil)
+		if err != nil {
+			return nil, err
```
Comment (HaoyangLiu) [inline l2geth/consensus/clique/synchronizer/server.go:104]: if an error happens, the retry will be aborted.

File: l2geth/consensus/clique/synchronizer/server.go [hunk 7]
```diff
@@ -88,9 +98,20 @@ func (sync *Synchronizer) GetSequencerSet() (SequencerSequencerInfos, error) {
...
+		}
+		time.Sleep(GET_SEQUENCER_TIMEOUT * time.Second)
+		if len(seqInfos) == 0 && i == 99 {
```
Comment (HaoyangLiu) [inline l2geth/consensus/clique/synchronizer/server.go:107]: The `99` means?

Comment (bradyjoestar) [reply l2geth/consensus/clique/synchronizer/server.go]: removed.

File: l2geth/consensus/clique/synchronizer/server.go (hunk 6 above)
Comment (bradyjoestar) [reply l2geth/consensus/clique/synchronizer/server.go]: > if an error happens, the retry will be aborted.

yeah, replace with continue.

File: l2geth/consensus/clique/synchronizer/server.go (hunk 5 above)
Comment (bradyjoestar) [reply l2geth/consensus/clique/synchronizer/server.go]: > We can use `common.Address{}.toString()` to replace `"0x0000000000000000000000000000000000000000"`

changed.

File: l2geth/eth/backend.go (hunk 4 above)
Comment (bradyjoestar) [reply l2geth/eth/backend.go]: > We can just revert the change to this file

reverted.

File: l2geth/internal/ethapi/api.go (hunk 3 above)
Comment (bradyjoestar) [reply l2geth/internal/ethapi/api.go]: > Can we just revert the changes to this file?

reverted.

**Review by HaoyangLiu — APPROVED** [review]
